synchronization/src/cyface/java/de/cyface/synchronization/CyfaceHttpConnection.java

```diff
@@ -199,13 +199,20 @@ private HttpResponse readResponseFromConnection(final HttpURLConnection con)
      * @return the {@link String} read from the InputStream
      */
     private String readInputStream(final InputStream inputStream) throws IOException {
-        final BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
-        final StringBuilder responseString = new StringBuilder();
-        String line;
-        while ((line = bufferedReader.readLine()) != null) {
-            responseString.append(line);
+        BufferedReader bufferedReader = null;
+        StringBuilder responseString;
+        try {
+            bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+            responseString = new StringBuilder();
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                responseString.append(line);
+            }
+        } finally {
+            if (bufferedReader != null) {
+                bufferedReader.close();
+            }
         }
-        bufferedReader.close();
         return responseString.toString();
```

Comment:
Please move the return statement into the try block. It is not necessary to run the finally before the return. Finally always runs at the end of a block, even if there was a return before.